### PR TITLE
Update Rust to 1.56/1.52 and use UBI in rust-toolset

### PR DESCRIPTION
**Pull request by vrutkovs**
Dedicated Rust UBI image was deprecated, we should be installing Rust for UBI

### CODE REVIEW

File: hack/e2e.sh [hunk 1]
```diff
@@ -74,7 +74,7 @@ backoff oc -n openshift-monitoring \
 backoff oc apply -f /tmp/cluster-monitoring-config.yaml
 
 # Wait for user workload monitoring is deployed
-backoff oc -n openshift-user-workload-monitoring wait --for=condition=Ready pod -l app=thanos-ruler
+#backoff oc -n openshift-user-workload-monitoring wait --for=condition=Ready pod -l app=thanos-ruler
```
Comment (LalatenduMohanty) [inline hack/e2e.sh]: It is not clear why we need this.

Comment (vrutkovs) [reply hack/e2e.sh]: Investigating why UWM verification doesn't work as expected. I'll move this to a separate PR

**Review by PratikMahajan — APPROVED** [review]
/lgtm

**Comment by vrutkovs** [comment]
/hold

Still no luck passing e2e

**Review by PratikMahajan — APPROVED** [review]
/lgtm

**Pull request merged**